sdk/core/azure-core/src/test/java/com/azure/core/UserAgentTests.java

```diff
@@ -57,4 +58,52 @@ public void customUserAgentTests() throws Exception {
                 new URL("http://localhost"))).block();
         Assertions.assertEquals(200, response.getStatusCode());
     }
+
+    @Test
+    public void defaultApplicationIdUserAgentTest() throws Exception {
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(new NoOpHttpClient() {
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    String header = request.getHeaders().getValue("User-Agent");
+                    String expectedHeaderPrefix = "azsdk-java-package.name";
+                    Assertions.assertTrue(header.startsWith(expectedHeaderPrefix));
+                    return Mono.just(new MockHttpResponse(request, 200));
+                }
+            })
+            .policies(new UserAgentPolicy(
+                null,
+                "package.name",
+                "package_version",
+                Configuration.NONE,
+                () -> "1.0"))
+            .build();
+        HttpResponse response = pipeline.send(new HttpRequest(HttpMethod.GET,
+            new URL("http://localhost"))).block();
+        Assertions.assertEquals(200, response.getStatusCode());
+    }
+
+    @Test
+    public void customApplicationIdUserAgentTest() throws Exception {
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(new NoOpHttpClient() {
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    String header = request.getHeaders().getValue("User-Agent");
+                    String expectedHeaderPrefix = "user_specified_appId";
+                    Assertions.assertTrue(header.startsWith(expectedHeaderPrefix));
```

Comment:
Could we validate the whole User-Agent string

Reply:
added